bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/Clip2BridgeHandler.java

```diff
@@ -0,0 +1,738 @@
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingRegistry;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.ThingTypeUID;
+import org.openhab.core.thing.ThingUID;
+import org.openhab.core.thing.binding.BaseBridgeHandler;
+import org.openhab.core.thing.binding.ThingHandler;
+import org.openhab.core.thing.binding.ThingHandlerService;
+import org.openhab.core.thing.binding.builder.BridgeBuilder;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.osgi.framework.Bundle;
+import org.osgi.framework.FrameworkUtil;
+import org.osgi.framework.ServiceRegistration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Bridge handler for a CLIP 2 bridge. It communicates with the bridge via CLIP 2 end points, and reads and writes API
+ * V2 resource objects. It also subscribes to the server's SSE event stream, and receives SSE events from it.
+ *
+ * @author Andrew Fiddian-Green - Initial contribution.
+ */
+@NonNullByDefault
+public class Clip2BridgeHandler extends BaseBridgeHandler {
+
+    public static final Set<ThingTypeUID> SUPPORTED_THING_TYPES = Set.of(THING_TYPE_BRIDGE_API2);
+
+    private static final int FAST_SCHEDULE_MILLI_SECONDS = 500;
+    private static final int APPLICATION_KEY_MAX_TRIES = 600; // i.e. 300 seconds, 5 minutes
+    private static final int RECONNECT_DELAY_SECONDS = 10;
+    private static final int RECONNECT_MAX_TRIES = 5;
+
+    private static final ResourceReference DEVICE = new ResourceReference().setType(ResourceType.DEVICE);
+    private static final ResourceReference ROOM = new ResourceReference().setType(ResourceType.ROOM);
+    private static final ResourceReference ZONE = new ResourceReference().setType(ResourceType.ZONE);
+    private static final ResourceReference BRIDGE = new ResourceReference().setType(ResourceType.BRIDGE);
+    private static final ResourceReference BRIDGE_HOME = new ResourceReference().setType(ResourceType.BRIDGE_HOME);
+    private static final ResourceReference SCENE = new ResourceReference().setType(ResourceType.SCENE);
+
+    /**
+     * List of resource references that need to be mass down loaded.
+     * NOTE: the SCENE resources must be mass down loaded first!
+     */
+    private static final List<ResourceReference> MASS_DOWNLOAD_RESOURCE_REFERENCES = Arrays.asList(SCENE, DEVICE, ROOM,
+            ZONE);
+
+    private final Logger logger = LoggerFactory.getLogger(Clip2BridgeHandler.class);
+
+    private final HttpClientFactory httpClientFactory;
+    private final ThingRegistry thingRegistry;
+    private final Bundle bundle;
+    private final LocaleProvider localeProvider;
+    private final TranslationProvider translationProvider;
+    private final ExecutorService threadPool = ThreadPoolManager.getPool("hue-api2-bridge");
```

Comment:
Why create own thread pool? See https://github.com/openhab/openhab-addons/pull/13570#discussion_r1240944382.

If this is a work-around to this reported issue: https://community.openhab.org/t/hue-binding-for-api-v2-3-4-0-0-4-0-0-0/146799/11

then probably it is not the right solution, since then all bindings would have to resort to using own thread pools to work around such issues.

See also https://www.openhab.org/docs/developer/guidelines.html#e-runtime-behavior

Reply:
Yes. I did it to make the guy happy. But I think you asking me to make him unhappy again? Or??

Reply:
If the reason for violating the guidelines is that one user reported a 5 second delay, and that user has >20 bindings installed and >6000 lines of rules, then I'd suggest to either
- Help pinpointing the root cause for that user. That of course requires quite some effort, also from the user.
- Or find out if any other users have the problem too.

I don't think one such user report justifies the provided work-around.

@fwolter - do you have any additional recommendations here?

Reply:
Thing specific thread pools should only be created if there are long running/never ending tasks like a server for a low level binary protocol or waiting for data on a serial interface, which would otherwise occupy a thread in the framework's thread pool for the Thing's lifetime.

Waiting for (HTTP) IO should be done in the framework's thread pool. Also, if there are two HTTP requests in a row.

I don't quite understand how a custom thread pool vs using the framework's thread pool fixes a 5s delay. It could be a coincidence that the problem disappeared after changing the code. But it doesn't sound like a solution.

Reply:
> I don't quite understand how a custom thread pool vs using the framework's thread pool fixes a 5s delay

@fwolter his problem was that something else in his system is blocking the OH core thread pool. So implementing a binding specific thread pool eliminated the blocked OH core thread pool from the picture.

@jlaur @fwolter anyway I shall revert this binding to using OH core thread pool. But perhaps in exchange you can help the guy on the forum solve his problem..

https://community.openhab.org/t/hue-binding-for-api-v2-3-4-0-0-4-0-0-0/146799/23?u=andrewfg

Reply:
Ok. Done.